tools/build_pak_files.py

```diff
@@ -0,0 +1,129 @@
+
+            # create symlink
+            print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_dir} -> {content_dir}")
+            os.symlink(content_dir, unreal_project_content_dir)
+
+            for platform in args.platforms:
+
+                cmd = [
+                    unreal_editor_bin,
+                    uproject,
+                    "-run=Cook",
+                    "-TargetPlatform=" + platform + "NoEditor",
+                    "-fileopenlog",
+                    "-ddc=InstalledDerivedDataBackendGraph",
+                    "-unversioned",
+                    "-stdout",
+                    "-FullStdOutLogOutput",
+                    "-CrashForUAT",
+                    "-unattended",
+                    "-NoLogTimes",
+                    "-UTF8Output"
+                ]
+                print(f"[SPEAR | build_pak_files.py] Executing: {' '.join(cmd)}")
+                cmd_result = subprocess.run(cmd, check=True)
+
+                platform_dir = os.path.realpath(os.path.join(unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor"))
+                content_dirs = [
+                    os.path.join(platform_dir, "Engine", "Content", "Animation"),
+                    os.path.join(platform_dir, "Engine", "Content", "BasicShapes"),
+                    os.path.join(platform_dir, "Engine", "Content", "EngineResources"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Math"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Texturing"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "Engine_MaterialFunctions02", "Utility"),
+                    os.path.join(platform_dir, "Engine", "Content", "Functions", "MaterialLayerFunctions"),
+                    os.path.join(platform_dir, project,  "Content")
+                ]
+
+                # create the output_dir
+                os.makedirs(args.output_dir, exist_ok=True)
+
+                file_name = f"Map_{os.path.basename(content_dir)}_{platform}"
+
+                # construct path to the output pak file
+                pak_file = os.path.join(args.output_dir, file_name + ".pak")
+
+                # text file used to generate the final pak file
+                txt_path = os.path.join(args.output_dir, file_name + ".txt")
+
+                for i, content_dir in enumerate(content_dirs):
+                    with open(txt_path, mode="w" if i==0 else "a") as f:
+                        for content_file in glob.glob(os.path.join(content_dir, "**", "*.*"), recursive=True):
+                            assert content_file.startswith(platform_dir)
+                            content_file = content_file.replace('\\', "/")
+                            mount_file = posixpath.join("..", "..", f"..{content_file.split(platform + 'NoEditor')[1]}")
```

Comment:
Why are we setting `mount_file` to `../../..path/to/uasset` (note the missing `/` before `path`)?

Reply:
The output of `content_file.split(platform + 'NoEditor')[1]` has a trailing forward slash (`/`) such as `/SpearSim/Content/....`.